source/distributed/tria.cc

```diff
@@ -1935,21 +1935,27 @@ namespace parallel
             int dummy = 0;
             unsigned int req_counter = 0;
 
-            for (std::set<unsigned int>::iterator it = this->number_cache.level_ghost_owners.begin();
+            for (std::set<types::subdomain_id>::iterator it = this->number_cache.level_ghost_owners.begin();
                  it != this->number_cache.level_ghost_owners.end();
                  ++it, ++req_counter)
               {
-                MPI_Isend(&dummy, 1, MPI_INT,
+                Assert (typeid(types::subdomain_id)
```

Comment:
What is the compiler support for typeid? Do you need the typeinfo header? Maybe it would be easier to just do a sizeof() comparison.